ocrd_zip.md

```diff
@@ -127,25 +130,26 @@ be referenced in a `mets:file/mets:Flocat` in the `mets.xml`.
 To pack a workspace to OCRD-ZIP:
 
 * Create a temporary folder `TMP`
-* Copy mets.xml to `TMP/data/mets.xml`
-* Foreach `mets:file` `f` in `TMP/data/mets.xml`:
-  * If it is not a `file://`-URL
-    * If `Ocrd-Manifestation-Depth` is `partial`
+* Foreach `mets:file` `f` in the source METS:
+  * If it is not a `file://`-URL and not a file path
+    * If `Ocrd-Manifestation-Depth` is `partial`,
       continue
   * Download/Copy the file to a location within `TMP/data`. The structure SHOULD be `<USE>/<ID>` where
     * `<USE>` is the `USE` attribute of the parent `mets:fileGrp`
     * `<ID>` is the `ID` attribute of the `mets:file`
-  * Replace the URL of `f` with `file:///data/<USE>/<ID>` in
-    * all `mets:FLocat` of `TMP/data/mets.xml`
+  * Replace the URL of `f` with the path relative to `/data` (SHOULD be `<USE>/<ID>`) in
+    * all `mets:FLocat` of the METS
     * all other files in the workspace, esp. PAGE-XML
+* Write out the changed METS to `TMP/data/mets.xml` (or another location in `TMP/data/`, specified by [`Ocrd-Mets`](#ocrd-mets)
 * Package `TMP` as a BagIt bag
 
 ### Unpacking OCRD-ZIP to a workspace
 
 * Unzip OCRD-ZIP `z` to a folder `TMP`
 * Foreach file `f` in `TMP/data/mets.xml`:
-  * If it is not a `file://`-URL, continue
-  * Replace the URL of `f` with `file://<ABSPATH>`, where `<ABSPATH>` is the absolute path to `f`, in
+  * If it is not a `file://`-URL and not a file path,
+    continue
+  * Replace the URL of `f` with `<ABSPATH>`, where `<ABSPATH>` is the absolute path to `f`, in
```

Comment:
I want `/data` to be effectively a workspace when unzipped. My takeaway from the feedback to my `url-sources.txt` proposal was that we should just allow relative paths.

At the moment, the `ocrd zip spill` implementation won't modify any `mets:FLocat` because they are either relative filenames at this point or HTTP-URL.

Get rid of the absolute paths for good?

Reply:
Relative files allowed in general METS too, now, see updated commits.